### ISSUE TITLE
Random failure in Unit.PointwiseFunctions.GeneralRelativity.GhQuantities

### ISSUE THREAD

**Issue by nilsvu** [sxs-collaboration/spectre#3060]
# Bug reports:

The following random-value test failed here https://github.com/sxs-collaboration/spectre/pull/3057/checks?check_run_id=2240041757:

```
-------------------------------------------------------------------------------
Unit.PointwiseFunctions.GeneralRelativity.GhQuantities
-------------------------------------------------------------------------------
/__w/spectre/spectre/tests/Unit/PointwiseFunctions/GeneralRelativity/Test_ComputeGhQuantities.cpp:777
...............................................................................

/__w/spectre/spectre/tests/Unit/Framework/TestingFramework.hpp:147: FAILED:
  CHECK( a == appx(b) )
with expansion:
  2.51099744427756377
  ==
  Approx( 2.51099744428211125 )
with messages:
  Seed is: 1340881727 from /__w/spectre/spectre/tests/Unit/Framework/
  CheckWithRandomValues.hpp:333
  function: gh_spatial_ricci_tensor
  /__w/spectre/spectre/tests/Unit/Framework/CheckWithRandomValues.hpp:221:
  result == pypp::call<ResultType>( module_name, function_name, std::get
  <ArgumentIs>(args)..., forward_to_pypp<ResultType>(std::get<MemberArgsIs>
  (member_args), used_for_size)...)
  a := T(0,0)=(2655.65,-3928.5,-10493,-5344.4,1040.78)
  T(1,0)=(-432.839,5564.7,7301.46,6058.86,-221.345)
  T(2,0)=(643.337,519.424,2.511,-6995.23,-262.965)
  T(1,1)=(9816.55,-3349.48,-15177.2,-9667.36,57.7929)
  T(2,1)=(-7779.06,3444.65,4912.65,-605.714,-5.28921)
  T(2,2)=(9605.68,-3039.09,-4390.85,4698.72,244.927)
  b := T(0,0)=(2655.65,-3928.5,-10493,-5344.4,1040.78)
  T(1,0)=(-432.839,5564.7,7301.46,6058.86,-221.345)
  T(2,0)=(643.337,519.424,2.511,-6995.23,-262.965)
  T(1,1)=(9816.55,-3349.48,-15177.2,-9667.36,57.7929)
  T(2,1)=(-7779.06,3444.65,4912.65,-605.714,-5.28921)
  T(2,2)=(9605.68,-3039.09,-4390.85,4698.72,244.927)
  a := (643.337,519.424,2.511,-6995.23,-262.965)
  b := (643.337,519.424,2.511,-6995.23,-262.965)

0.401 s: Unit.PointwiseFunctions.GeneralRelativity.GhQuantities
===============================================================================
test cases:     1 |     0 passed | 1 failed
assertions: 29973 | 29972 passed | 1 failed

A catch test has failed.
```

**Comment by nilsvu** [comment]
@prayush this is currently failing O(10%) of CI tests. It's caused by the recently merged #2529. Could you take a look please?

**Comment by nilsdeppe** [comment]
Another failure:
```
2021-04-01T00:08:13.0555375Z Filters: "Unit.PointwiseFunctions.GeneralRelativity.GhQuantities"
2021-04-01T00:08:13.0556577Z 
2021-04-01T00:08:13.0557053Z ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2021-04-01T00:08:13.0559920Z "Release.Unit.PointwiseFunctions.GeneralRelativity.GhQuantities" is a Catch v2.11.1 host application.
2021-04-01T00:08:13.0561719Z Run with -? for options
2021-04-01T00:08:13.0562015Z 
2021-04-01T00:08:13.0563423Z -------------------------------------------------------------------------------
2021-04-01T00:08:13.0564771Z Unit.PointwiseFunctions.GeneralRelativity.GhQuantities
2021-04-01T00:08:13.0566296Z -------------------------------------------------------------------------------
2021-04-01T00:08:13.0567413Z /__w/spectre/spectre/tests/Unit/PointwiseFunctions/GeneralRelativity/Test_ComputeGhQuantities.cpp:777
2021-04-01T00:08:13.0568383Z ...............................................................................
2021-04-01T00:08:13.0568654Z 
2021-04-01T00:08:13.0569317Z /__w/spectre/spectre/tests/Unit/Framework/TestingFramework.hpp:147: FAILED:
2021-04-01T00:08:13.0570020Z   CHECK( a == appx(b) )
2021-04-01T00:08:13.0570451Z with expansion:
2021-04-01T00:08:13.0570831Z   0.38578557065216046
2021-04-01T00:08:13.0571158Z   ==
2021-04-01T00:08:13.0571500Z   Approx( 0.38578557065375207 )
2021-04-01T00:08:13.0571895Z with messages:
2021-04-01T00:08:13.0572480Z   Seed is: 694217826 from /__w/spectre/spectre/tests/Unit/Framework/
2021-04-01T00:08:13.0573229Z   CheckWithRandomValues.hpp:333
2021-04-01T00:08:13.0573902Z   function: gh_spatial_ricci_tensor
2021-04-01T00:08:13.0574718Z   /__w/spectre/spectre/tests/Unit/Framework/CheckWithRandomValues.hpp:221:
2021-04-01T00:08:13.0575868Z   result == pypp::call<ResultType>( module_name, function_name, std::get
2021-04-01T00:08:13.0576754Z   <ArgumentIs>(args)..., forward_to_pypp<ResultType>(std::get<MemberArgsIs>
2021-04-01T00:08:13.0577460Z   (member_args), used_for_size)...)
2021-04-01T00:08:13.0578212Z   a := T(0,0)=(-0.327894,-7037.17,-365.987,-9.58758,74.4298)
2021-04-01T00:08:13.0578768Z   T(1,0)=(0.385786,8321.94,614.19,2.88387,-47.1463)
2021-04-01T00:08:13.0579299Z   T(1,1)=(-0.308559,-8184.77,-1456.94,-7.45279,13.2203)
2021-04-01T00:08:13.0580074Z   b := T(0,0)=(-0.327894,-7037.17,-365.987,-9.58758,74.4298)
2021-04-01T00:08:13.0580611Z   T(1,0)=(0.385786,8321.94,614.19,2.88387,-47.1463)
2021-04-01T00:08:13.0581142Z   T(1,1)=(-0.308559,-8184.77,-1456.94,-7.45279,13.2203)
2021-04-01T00:08:13.0581675Z   a := (0.385786,8321.94,614.19,2.88387,-47.1463)
2021-04-01T00:08:13.0582515Z   b := (0.385786,8321.94,614.19,2.88387,-47.1463)
2021-04-01T00:08:13.0582773Z 
2021-04-01T00:08:13.0583819Z 0.335 s: Unit.PointwiseFunctions.GeneralRelativity.GhQuantities
```
